docs/clickhouse-debugging.adoc

```diff
@@ -0,0 +1,199 @@
+:showtitle:
+:numbered:
+:toc: left
+
+= Omicron Clickhouse Debugging Guide
+
+This is a guide for debugging Clickhouse on a variety of environments.
+
+If you have advice that is not covered here, consider adding it!
+
+== Debugging on a Live System
+
+The following provides instructions for connecting to a Clickhouse shell on a running system.
+
+. **Find the zone running Clickhouse**. This can be accomplished by running `zoneadm list -cv`, and finding the zone with a prefix of `oxz_clickhouse`. If you're running on a multi-machine system (e.g., dogfood, colo, etc) and you have access to the `pilot` binary, you can ask all sleds at once for the location of Clickhouse with:
+// '+' for list continuation to insert code blocks while keeping the list order
++
+[source,bash]
+----
+# Run from the switch zone.
+$ pilot host exec -c "zoneadm list -c | grep clickhouse" 0-31
+----
+. **Log into that zone**. This can be done using:
++
+[source,bash]
+----
+# Run from the switch zone
+$ pilot host login <cubby where clickhouse exists>
+
+# Run from the machine with the Clichouse zone
+$ pfexec zlogin oxz_clickhouse_<UUID>
+----
+
+. **Identify the IP address of Clickhouse**. This is possible using `ipadm`:
++
+[source,bash]
+----
+# Run from within the Clickhouse zone
+$ ipadm
+ADDROBJ           TYPE     STATE        ADDR
+lo0/v4            static   ok           127.0.0.1/8
+lo0/v6            static   ok           ::1/128
+oxControlService8/ll addrconf ok        fe80::8:20ff:fe35:6b0a%oxControlService8/10
+oxControlService8/omicron6 static ok    fd00:1122:3344:107::4/64   <-- It's this one!
+----
+. **Log into Clickhouse using the CLI**
++
+[source,bash]
+----
+# Run from within the Clickhouse zone
+$ /opt/oxide/clickhouse/clickhouse client --host fd00:1122:3344:107::4
```

Comment:
Can this process be automated?

Reply:
I want it to be. I'm kinda curious if omdb can handle a lot of this stuff, or if `oxdb` can help here - I know that's a tool that can be used to query clickhouse, but I didn't see it in any of the packaging stuff, so I wasn't sure if it was available on deployed racks.

But anyway, yeah, my dream would be:
- omdb has a command to find clickhouse via DNS
- omdb can then either call into oxdb, or re-use that functionality as a library...
- ... and as a caller, running something like `omdb clickhouse query <timeseries_name>` and seeing all data would be rad